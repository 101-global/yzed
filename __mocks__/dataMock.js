export const model = {
  name: 'YZED',
  brandName: 'YZED',
  imageSlug: 'yzed',
  glbFile: 'https://www.testYZEDfile.com/glbFile.glb',
  usdzFile: 'https://www.testYZEDfile.com/usdzFile.usdz',
  animatedGlbFile: 'https://www.testYZEDfile.com/animatedGlbFile.glb',
  exposure: 1,
  imageCount: 5,
};

export const buttonMock = {
  title: 'Button Title',
  href: 'https://www.button-link.com/to-the-site',
};
