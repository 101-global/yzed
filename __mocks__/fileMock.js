module.exports = 'File Sub Mock';
