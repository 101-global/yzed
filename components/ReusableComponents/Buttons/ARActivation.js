import React, { useContext, useEffect, useState } from 'react';

import theme from '../../../utilities/theme';

const ARActivation = () => {
  return (
    <>
      <button slot='ar-button' aria-label='activate ar' className='ar-button'>
        <div className='flex flex-row justify-center items-center p-1'>
          <svg
            aria-hidden='true'
            width='40'
            height='40'
            viewBox='0 0 40 40'
            fill='none'
            xmlns='http://www.w3.org/2000/svg'
            className='activate-ar-button'>
            <g clipPath='url(#clip0)'>
              <path
                d='M39.6653 20C39.6653 30.7696 30.9348 39.5 20.1653 39.5C9.39573 39.5 0.665283 30.7696 0.665283 20C0.665283 9.23045 9.39573 0.5 20.1653 0.5C30.9348 0.5 39.6653 9.23045 39.6653 20Z'
                className='fill-opposite stroke-main'
                strokeMiterlimit='10'
              />
              <path
                d='M14.8425 17.2466H11.4558L11.9991 15.6399H15.1325L14.8425 17.2466Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M18.2294 17.2466H14.8428L15.1328 15.6399H18.2661L18.2294 17.2466Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M21.6162 17.2466H18.2295L18.2662 15.6399H21.3995L21.6162 17.2466Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M25.0028 17.2466H21.6161L21.3994 15.6399H24.5327L25.0028 17.2466Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M28.3896 17.2466H25.003L24.533 15.6399H27.6663L28.3896 17.2466Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M14.5029 19.1332H10.8196L11.4562 17.2466H14.8429L14.5029 19.1332Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M18.1865 19.1332H14.5032L14.8432 17.2466H18.2298L18.1865 19.1332Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M21.8701 19.1332H18.1868L18.2301 17.2466H21.6168L21.8701 19.1332Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M25.5526 19.1332H21.8693L21.616 17.2466H25.0026L25.5526 19.1332Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M29.2367 19.1332H25.5534L25.0034 17.2466H28.3901L29.2367 19.1332Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M32.9993 27.4999H7.99927L11.9993 15.6399H27.6659L32.9993 27.4999Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M14.0994 21.3866H10.0627L10.8194 19.1333H14.5027L14.0994 21.3866Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M18.136 21.3866H14.0994L14.5027 19.1333H18.186L18.136 21.3866Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M22.176 21.3866H18.136L18.186 19.1333H21.8693L22.176 21.3866Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M26.2127 21.3866H22.1761L21.8694 19.1333H25.5527L26.2127 21.3866Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M30.2496 21.3866H26.213L25.553 19.1333H29.2363L30.2496 21.3866Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M13.6099 24.1167H9.13989L10.0632 21.3867H14.0999L13.6099 24.1167Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M18.0763 24.1167H13.6096L14.0996 21.3867H18.1363L18.0763 24.1167Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M22.5431 24.1167H18.0764L18.1364 21.3867H22.1764L22.5431 24.1167Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M27.0098 24.1167H22.5432L22.1765 21.3867H26.2132L27.0098 24.1167Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M31.4798 24.1167H27.0098L26.2131 21.3867H30.2498L31.4798 24.1167Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M12.9995 27.4998H7.99951L9.13951 24.1165H13.6095L12.9995 27.4998Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M18.0002 27.4998H13.0002L13.6102 24.1165H18.0769L18.0002 27.4998Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M22.9998 27.4998H17.9998L18.0764 24.1165H22.5431L22.9998 27.4998Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M27.9999 27.4998H22.9999L22.5432 24.1165H27.0099L27.9999 27.4998Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M32.9998 27.4998H27.9998L27.0098 24.1165H31.4798L32.9998 27.4998Z'
                className='stroke-main'
                strokeWidth='0.5'
                strokeMiterlimit='10'
              />
              <path
                d='M20.1558 23.3434C23.5008 23.3434 26.2125 22.3823 26.2125 21.1967C26.2125 20.0111 23.5008 19.05 20.1558 19.05C16.8108 19.05 14.0991 20.0111 14.0991 21.1967C14.0991 22.3823 16.8108 23.3434 20.1558 23.3434Z'
                className='fill-main'
              />
              <path
                d='M20.1563 19.9998C23.1828 19.9998 25.6363 17.5747 25.6363 14.5832C25.6363 11.5916 23.1828 9.1665 20.1563 9.1665C17.1297 9.1665 14.6763 11.5916 14.6763 14.5832C14.6763 17.5747 17.1297 19.9998 20.1563 19.9998Z'
                className='fill-opposite stroke-main'
                strokeMiterlimit='10'
              />
            </g>
            <defs>
              <clipPath id='clip0'>
                <rect width='40' height='40' className='fill-opposite' />
              </clipPath>
            </defs>
          </svg>
          <p className='ml-2 font-light text-sm tracking-wide'>View AR</p>
        </div>
      </button>

      <style jsx global>{`
        button.ar-button {
          position: absolute;
          bottom: 0;
          right: 0;
        }
        .dark-mode svg.activate-ar-button .stroke-main {
          stroke: ${theme.colors.white};
        }
        .dark-mode svg.activate-ar-button .fill-main {
          fill: ${theme.colors.white};
        }
        .dark-mode svg.activate-ar-button .fill-opposite {
          fill: ${theme.colors.black};
        }
        .light-mode svg.activate-ar-button .stroke-main {
          stroke: ${theme.colors.black};
        }
        .light-mode svg.activate-ar-button .fill-main {
          fill: ${theme.colors.black};
        }
        .light-mode svg.activate-ar-button .fill-opposite {
          fill: ${theme.colors.white};
        }
      `}</style>
    </>
  );
};

export default ARActivation;
