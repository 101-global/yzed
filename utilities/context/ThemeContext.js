import React, { useCallback, useEffect, useState } from 'react';

import PropTypes from 'prop-types';

import useDarkMode from 'use-dark-mode';

export const ThemeContext = React.createContext();

const ThemeProvider = ({ children }) => {
  const darkMode = useDarkMode();
  console.log(darkMode);
  const toggleDarkMode = () => {
    console.log("CLICK")
    darkMode.toggle();
  };
  return (
    <ThemeContext.Provider value={{ darkMode, toggleDarkMode }}>{children}</ThemeContext.Provider>
  );
};

ThemeProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default ThemeProvider;
