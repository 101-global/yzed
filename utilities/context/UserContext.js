import React, { useCallback, useEffect, useState } from 'react';
import { defaultIcon, tokenName } from '../constants';
import { fbData, googleData } from '../dataHelpers';

import PropTypes from 'prop-types';
import cookie from 'js-cookie';
import firebase from '../firebaseSetup';
import { server } from '../../config/index';
import { signupStates } from '../enums';
import { useRouter } from 'next/router';

export const UserContext = React.createContext();

const dbh = firebase.firestore();

const googleProvider = new firebase.auth.GoogleAuthProvider();
const fbProvider = new firebase.auth.FacebookAuthProvider();

fbProvider.addScope('email');

const UserProvider = ({ children }) => {
  const [userLoading, setUserLoading] = useState(false);
  const [userError, setUserError] = useState(null);

  const router = useRouter();

  const setError = (error) => {
    setUserError(error);
    setUserLoading(false);
  };

  const refreshUserData = (route) => {
    setTimeout(() => {
      if (route) {
        router.push(route);
      } else {
        router.reload();
      }
    }, 2000);
  };

  const createUserDB = async (userID, email, firstName, lastName, emailVerified, route) => {
    await dbh
      .collection('users')
      .doc(userID)
      .set({
        email,
        firstName,
        lastName,
        profilePicture: firstName.slice(0, 1),
        emailVerified,
        role: 'USER',
      })
      .then((result) => {
        refreshUserData(route);
      })
      .catch((err) => {
        setError(err.message);
      });
  };

  const updateUserDB = async (userID, email, firstName, lastName, profilePicture) => {
    await dbh
      .collection('users')
      .doc(userID)
      .update({ email, firstName, lastName, profilePicture })
      .then((result) => {
        refreshUserData();
      })
      .catch((err) => {
        setError(err.message);
      });
  };

  const emailSignup = async (email, password, firstName, lastName, redirectPath) => {
    setUserLoading(true);
    await firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async (user) => {
        if (user) {
          const userID = user.user.uid;
          const emailVerified = user.user.emailVerified;
          const currentUser = await firebase.auth().currentUser;
          console.log(currentUser);
          await currentUser
            .sendEmailVerification()
            .then(async (resp) => {
              await createUserDB(
                userID,
                email,
                firstName,
                lastName,
                emailVerified,
                '/signup/confirm'
              );
            })
            .catch((err) => {
              setError(err.message);
            });
        }
      })
      .catch((err) => {
        console.log(err);
        setError(err);
      });
  };

  const googleSignup = async () => {
    setUserLoading(true);
    await firebase
      .auth()
      .signInWithPopup(googleProvider)
      .then(async (result) => {
        const data = googleData(result);
        const { userID, email, firstName, lastName, profilePicture } = data;
        await createUserDB(
          userID,
          email,
          firstName,
          lastName,
          profilePicture,
          true,
          '/signup/success'
        );
      })
      .catch((err) => {
        setError(err.message);
      });
  };

  const fbSignup = async () => {
    setUserLoading(true);
    await firebase
      .auth()
      .signInWithPopup(fbProvider)
      .then(async (result) => {
        const data = fbData(result);
        const { userID, email, firstName, lastName, profilePicture } = data;
        data.url;
        await createUserDB(
          userID,
          email,
          firstName,
          lastName,
          profilePicture,
          true,
          '/signup/success'
        );
      })
      .catch((err) => {
        setError(err.message);
      });
  };

  const emailLogin = async (email, password, redirectPath) => {
    setUserLoading(true);
    await firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(() => {
        refreshUserData('/');
      })
      .catch((err) => {
        console.log(err);
        setError(err);
      });
  };

  const googleLogin = () => {
    setUserLoading(true);
    firebase
      .auth()
      .signInWithPopup(googleProvider)
      .then(async (result) => {
        const data = googleData(result);
        const { userID, email, firstName, lastName, profilePicture } = data;
        await updateUserDB(userID, email, firstName, lastName, profilePicture);
      })
      .catch((err) => {
        setError(err.message);
      });
  };

  const fbLogin = () => {
    setUserLoading(true);
    firebase
      .auth()
      .signInWithPopup(fbProvider)
      .then(async (result) => {
        const data = fbData(result);
        const { userID, email, firstName, lastName, profilePicture } = data;
        data.url;
        await updateUserDB(userID, email, firstName, lastName, profilePicture);
      })
      .catch((err) => {
        setError(err.message);
      });
  };

  const logout = () => {
    if (window.confirm('Are you sure you want to log out?')) {
      firebase
        .auth()
        .signOut()
        .then(() => {
          router.reload();
        });
    }
  };

  const requestForgottenPasswordEmail = (email, callback) => {
    setUserLoading(true);
    const actionCodeSettings = {
      // After password reset, the user will be give the ability to go back
      // to this page.
      url: `${server}/login`,
      handleCodeInApp: true,
    };
    firebase
      .auth()
      .sendPasswordResetEmail(email, actionCodeSettings)
      .then(() => {
        callback(true);
        setUserLoading(false);
      })
      .catch((err) => {
        console.log(err);
        setError(err);
        setUserLoading(false);
      });
  };

  const resetPassword = (password, oobCode, callback) => {
    firebase
      .auth()
      .verifyPasswordResetCode(oobCode)
      .then((email) => {
        firebase
          .auth()
          .confirmPasswordReset(oobCode, password)
          .then((resp) => {
            callback(true);
            cookie.remove(tokenName);
            setTimeout(() => {
              emailLogin(email, password);
            }, 1000);
          })
          .catch((err) => {
            console.log(err.message);
            setError(error);
          });
      })
      .catch(function (error) {
        setError(error);
      });
  };

  const updatePassword = (password, successCallback) => {
    setUserLoading(true);
    const user = firebase.auth().currentUser;
    user
      .updatePassword(password)
      .then(() => {
        successCallback(true);
        setUserLoading(false);
      })
      .catch((err) => {
        setUserError(err);
        setUserLoading(false);
      });
  };

  const requestEmailVerification = (callback) => {
    const actionCodeSettings = {
      // After password reset, the user will be give the ability to go back
      // to this page.
      url: `${server}/login`,
      handleCodeInApp: true,
    };
    firebase
      .auth()
      .currentUser.sendEmailVerification(actionCodeSettings)
      .then((resp) => {
        console.log(resp);
      })
      .catch((err) => {
        setError(err.message);
      });
  };

  const confirmEmailVerification = (oobCode, callback) => {
    firebase
      .auth()
      .applyActionCode(oobCode)
      .then((resp) => {
        callback(false);
        // Email address has been verified.
        // TODO: Display a confirmation message to the user.
        // You could also provide the user with a link back to the app.
        // TODO: If a continue URL is available, display a button which on
        // click redirects the user back to the app via continueUrl with
        // additional state determined from that URL's parameters.
      })
      .catch((err) => {
        callback(false);
        setError(err.message);
      });
  };

  const onAuthStateChange = () => {
    return firebase.auth().onAuthStateChanged(async (user) => {
      if (user) {
        const token = await user.getIdToken();
        cookie.set(tokenName, token, { expires: 14 });
      } else {
        cookie.remove(tokenName);
      }
    });
  };

  useEffect(() => {
    const unsubscribe = onAuthStateChange();
    return () => {
      unsubscribe();
    };
  }, []);

  const requestEmailSignup = useCallback((username, password, firstName, lastName) => {
    emailSignup(username, password, firstName, lastName);
  });

  const requestGoogleSignup = useCallback(() => {
    googleSignup();
  });

  const requestFbSignup = useCallback(() => {
    fbSignup();
  });

  const requestEmailLogin = useCallback((email, password) => {
    emailLogin(email, password);
  });

  const requestGoogleLogin = useCallback(() => {
    googleLogin();
  });

  const requestFbLogin = useCallback(() => {
    fbLogin();
  });

  const requestLogout = useCallback(() => {
    logout();
  });

  return (
    <UserContext.Provider
      value={{
        userLoading,
        userError,
        requestEmailSignup,
        requestGoogleSignup,
        requestFbSignup,
        requestEmailLogin,
        requestGoogleLogin,
        requestFbLogin,
        requestLogout,
        requestForgottenPasswordEmail,
        requestEmailVerification,
        resetPassword,
        setUserError,
        confirmEmailVerification,
        updatePassword,
      }}>
      {children}
    </UserContext.Provider>
  );
};

UserProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default UserProvider;
