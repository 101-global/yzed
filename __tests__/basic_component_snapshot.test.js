import ARActivation from '../components/ReusableComponents/Buttons/ARActivation';
import LoadingBars from '../components/ReusableComponents/Loading/LoadingBars';
import LoadingSpinner from '../components/ReusableComponents/Loading/LoadingSpinner';
import React from 'react';
import { ThemeContext } from '../utilities/context/ThemeContext';
import ToggleDarkModeButton from '../components/ReusableComponents/Buttons/ToggleDarkMode';
import { buttonMock } from '../__mocks__/dataMock';
import renderer from 'react-test-renderer';

it('renders AR Button unchanged', () => {
  const tree = renderer.create(<ARActivation />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Loading Bars unchanged', () => {
  const tree = renderer.create(<LoadingBars />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Loading Spinner unchanged', () => {
  const tree = renderer.create(<LoadingSpinner />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Toggle Dark Mode Button unchanged', () => {
  const tree = renderer
    .create(
      <ThemeContext.Provider value={{ darkMode: false, toggleDarkMode: buttonMock }}>
        <ToggleDarkModeButton />
      </ThemeContext.Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
