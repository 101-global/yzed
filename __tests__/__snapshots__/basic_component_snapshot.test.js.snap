// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AR Button unchanged 1`] = `
<button
  aria-label="activate ar"
  className="jsx-3516056933 ar-button"
  slot="ar-button"
>
  <div
    className="jsx-3516056933 flex flex-row justify-center items-center p-1"
  >
    <svg
      className="jsx-3516056933 activate-ar-button"
      fill="none"
      height="40"
      viewBox="0 0 40 40"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        className="jsx-3516056933"
        clip-path="url(#clip0)"
      >
        <path
          className="jsx-3516056933 fill-opposite stroke-main"
          d="M39.6653 20C39.6653 30.7696 30.9348 39.5 20.1653 39.5C9.39573 39.5 0.665283 30.7696 0.665283 20C0.665283 9.23045 9.39573 0.5 20.1653 0.5C30.9348 0.5 39.6653 9.23045 39.6653 20Z"
          stroke-miterlimit="10"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M14.8425 17.2466H11.4558L11.9991 15.6399H15.1325L14.8425 17.2466Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M18.2294 17.2466H14.8428L15.1328 15.6399H18.2661L18.2294 17.2466Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M21.6162 17.2466H18.2295L18.2662 15.6399H21.3995L21.6162 17.2466Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M25.0028 17.2466H21.6161L21.3994 15.6399H24.5327L25.0028 17.2466Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M28.3896 17.2466H25.003L24.533 15.6399H27.6663L28.3896 17.2466Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M14.5029 19.1332H10.8196L11.4562 17.2466H14.8429L14.5029 19.1332Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M18.1865 19.1332H14.5032L14.8432 17.2466H18.2298L18.1865 19.1332Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M21.8701 19.1332H18.1868L18.2301 17.2466H21.6168L21.8701 19.1332Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M25.5526 19.1332H21.8693L21.616 17.2466H25.0026L25.5526 19.1332Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M29.2367 19.1332H25.5534L25.0034 17.2466H28.3901L29.2367 19.1332Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M32.9993 27.4999H7.99927L11.9993 15.6399H27.6659L32.9993 27.4999Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M14.0994 21.3866H10.0627L10.8194 19.1333H14.5027L14.0994 21.3866Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M18.136 21.3866H14.0994L14.5027 19.1333H18.186L18.136 21.3866Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M22.176 21.3866H18.136L18.186 19.1333H21.8693L22.176 21.3866Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M26.2127 21.3866H22.1761L21.8694 19.1333H25.5527L26.2127 21.3866Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M30.2496 21.3866H26.213L25.553 19.1333H29.2363L30.2496 21.3866Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M13.6099 24.1167H9.13989L10.0632 21.3867H14.0999L13.6099 24.1167Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M18.0763 24.1167H13.6096L14.0996 21.3867H18.1363L18.0763 24.1167Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M22.5431 24.1167H18.0764L18.1364 21.3867H22.1764L22.5431 24.1167Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M27.0098 24.1167H22.5432L22.1765 21.3867H26.2132L27.0098 24.1167Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M31.4798 24.1167H27.0098L26.2131 21.3867H30.2498L31.4798 24.1167Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M12.9995 27.4998H7.99951L9.13951 24.1165H13.6095L12.9995 27.4998Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M18.0002 27.4998H13.0002L13.6102 24.1165H18.0769L18.0002 27.4998Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M22.9998 27.4998H17.9998L18.0764 24.1165H22.5431L22.9998 27.4998Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M27.9999 27.4998H22.9999L22.5432 24.1165H27.0099L27.9999 27.4998Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 stroke-main"
          d="M32.9998 27.4998H27.9998L27.0098 24.1165H31.4798L32.9998 27.4998Z"
          stroke-miterlimit="10"
          stroke-width="0.5"
        />
        <path
          className="jsx-3516056933 fill-main"
          d="M20.1558 23.3434C23.5008 23.3434 26.2125 22.3823 26.2125 21.1967C26.2125 20.0111 23.5008 19.05 20.1558 19.05C16.8108 19.05 14.0991 20.0111 14.0991 21.1967C14.0991 22.3823 16.8108 23.3434 20.1558 23.3434Z"
        />
        <path
          className="jsx-3516056933 fill-opposite stroke-main"
          d="M20.1563 19.9998C23.1828 19.9998 25.6363 17.5747 25.6363 14.5832C25.6363 11.5916 23.1828 9.1665 20.1563 9.1665C17.1297 9.1665 14.6763 11.5916 14.6763 14.5832C14.6763 17.5747 17.1297 19.9998 20.1563 19.9998Z"
          stroke-miterlimit="10"
        />
      </g>
      <defs
        className="jsx-3516056933"
      >
        <clipPath
          className="jsx-3516056933"
          id="clip0"
        >
          <rect
            className="jsx-3516056933 fill-opposite"
            height="40"
            width="40"
          />
        </clipPath>
      </defs>
    </svg>
    <p
      className="jsx-3516056933 ml-2 font-light text-sm tracking-wide"
    >
      View AR
    </p>
  </div>
</button>
`;

exports[`renders Loading Bars unchanged 1`] = `
<div
  className="jsx-2898363057 spinner-container flex flex-col items-center justify-center bg-black"
>
  <div
    className="jsx-2898363057 lds-facebook"
  >
    <div
      className="jsx-2898363057"
    />
    <div
      className="jsx-2898363057"
    />
    <div
      className="jsx-2898363057"
    />
  </div>
</div>
`;

exports[`renders Loading Spinner unchanged 1`] = `
<div
  className="jsx-1134652233 spinner-container flex flex-col items-center justify-center"
>
  <div
    className="jsx-1134652233 lds-ellipsis"
  >
    <div
      className="jsx-1134652233"
    />
    <div
      className="jsx-1134652233"
    />
    <div
      className="jsx-1134652233"
    />
    <div
      className="jsx-1134652233"
    />
  </div>
</div>
`;

exports[`renders Toggle Dark Mode Button unchanged 1`] = `
<div
  className="jsx-2092124997 jsx-996897847 toggle-container"
>
  <input
    aria-label="toggle dark mode"
    className="jsx-2092124997 jsx-996897847 toggle"
    onClick={[Function]}
    type="checkbox"
  />
  <svg
    className="toggle-icon moon"
    fill="none"
    height="20"
    viewBox="0 0 19 20"
    width="19"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7.81751 1.68458C6.97867 4.74283 7.56411 7.96419 9.42528 10.5244C11.2835 13.0817 14.1612 14.6341 17.333 14.7914C16.7709 15.6244 16.0719 16.3496 15.2534 16.9438C13.7709 18.0215 12.0204 18.5894 10.1942 18.5894C7.20586 18.5894 4.47091 17.0749 2.87771 14.5409C0.626253 10.9613 1.34858 6.18749 4.5612 3.43506C5.51945 2.6137 6.61169 2.02535 7.80586 1.68166C7.81169 1.68458 7.81751 1.68458 7.81751 1.68458ZM7.81751 0.810791C7.73596 0.810791 7.65149 0.822441 7.56702 0.845742C6.29712 1.20982 5.07674 1.84768 3.99324 2.77681C0.454409 5.80885 -0.340737 11.0661 2.14082 15.0127C3.96994 17.9195 7.06023 19.469 10.1971 19.469C12.1311 19.469 14.0855 18.8807 15.769 17.6545C16.701 16.9788 17.4728 16.1632 18.0758 15.2574C18.4515 14.6982 18.0495 13.9467 17.3767 13.9147C14.5952 13.7865 11.8952 12.4409 10.133 10.0147C8.368 7.58555 7.92528 4.60302 8.66217 1.9205C8.81945 1.33506 8.37382 0.810791 7.81751 0.810791Z"
      fill="#fff"
    />
  </svg>
  <svg
    className="toggle-icon sun"
    fill="none"
    height="21"
    viewBox="0 0 21 21"
    width="21"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10.263 15.16C7.86298 15.16 5.90869 13.2086 5.90869 10.8057C5.90869 8.40285 7.86298 6.45142 10.263 6.45142C12.663 6.45142 14.6173 8.40571 14.6173 10.8057C14.6173 13.2057 12.663 15.16 10.263 15.16ZM10.263 7.15999C8.25156 7.15999 6.61727 8.79714 6.61727 10.8057C6.61727 12.8143 8.25441 14.4514 10.263 14.4514C12.2716 14.4514 13.9087 12.8143 13.9087 10.8057C13.9087 8.79714 12.2744 7.15999 10.263 7.15999Z"
      fill="#272727"
    />
    <path
      d="M10.6139 0.805664H9.9082V4.30854H10.6139V0.805664Z"
      fill="#272727"
    />
    <path
      d="M5.56769 1.96941L4.95654 2.32227L6.70798 5.35575L7.31913 5.00289L5.56769 1.96941Z"
      fill="#272727"
    />
    <path
      d="M1.77913 5.50164L1.42627 6.11279L4.45975 7.86423L4.81261 7.25308L1.77913 5.50164Z"
      fill="#272727"
    />
    <path
      d="M3.76507 10.4512H0.262207V11.1569H3.76507V10.4512Z"
      fill="#272727"
    />
    <path
      d="M4.46073 13.7477L1.42725 15.4995L1.78018 16.1107L4.81366 14.3589L4.46073 13.7477Z"
      fill="#272727"
    />
    <path
      d="M6.70931 16.2558L4.95752 19.2893L5.56867 19.6422L7.32046 16.6087L6.70931 16.2558Z"
      fill="#272727"
    />
    <path
      d="M10.6139 17.3027H9.9082V20.8056H10.6139V17.3027Z"
      fill="#272727"
    />
    <path
      d="M13.8167 16.257L13.2056 16.6099L14.957 19.6433L15.5681 19.2905L13.8167 16.257Z"
      fill="#272727"
    />
    <path
      d="M16.0667 13.7497L15.7139 14.3608L18.7474 16.1123L19.1002 15.5011L16.0667 13.7497Z"
      fill="#272727"
    />
    <path
      d="M20.2621 10.4512H16.7593V11.1569H20.2621V10.4512Z"
      fill="#272727"
    />
    <path
      d="M18.7493 5.49822L15.7158 7.25L16.0687 7.86115L19.1022 6.10937L18.7493 5.49822Z"
      fill="#272727"
    />
    <path
      d="M14.9578 1.96749L13.2061 5.00098L13.8172 5.35391L15.569 2.32042L14.9578 1.96749Z"
      fill="#272727"
    />
    <path
      d="M10.2619 14.4516C12.2754 14.4516 13.9077 12.8194 13.9077 10.8059C13.9077 8.7924 12.2754 7.16016 10.2619 7.16016C8.24846 7.16016 6.61621 8.7924 6.61621 10.8059C6.61621 12.8194 8.24846 14.4516 10.2619 14.4516Z"
      fill="transparent"
    />
  </svg>
</div>
`;
